src/AssetsPage/AssetsTable/index.jsx

```diff
@@ -52,37 +65,97 @@ class AssetsTable extends React.Component {
     });
   }
 
+  modalClose() {
+    this.setState({ modalOpen: false });
+  }
+
+  deleteAsset() {
+    this.props.deleteAsset(this.props.assetsParameters, this.state.assetToDelete.id);
+    this.setState({ modalOpen: false });
+  }
+
+  renderModal() {
+    return (
+      <Modal
+        open={this.state.modalOpen}
+        title={`Delete ${this.state.assetToDelete.display_name}`}
+        body={this.renderBody()}
+        closeText="Cancel"
+        onClose={this.modalClose}
+        buttons={[
+          <Button
+            display="Yes, delete."
+            buttonType="primary"
+            onClick={this.deleteAsset}
+          />,
+        ]}
+      />
+    );
+  }
+
+  renderBody() {
+    return (
+      <div>
+        <span className={classNames(FontAwesomeStyles.fa, FontAwesomeStyles['fa-exclamation-triangle'])} aria-hidden="true" />
+        Are you sure you wish to delete this item?
+        This cannot be reversed! Any content that links/refers to this item will no longer work
+        (e.g. images and/or links may break).
+      </div>
+    );
+  }
+
   render() {
     return (!this.props.assetsList.length) ? (
       <span>Loading....</span>
     ) : (
-      <Table
-        columns={[
-          {
-            label: 'Name',
-            key: 'display_name',
-          },
-          {
-            label: 'Type',
-            key: 'content_type',
-          },
-          {
-            label: 'Date Added',
-            key: 'date_added',
-          },
-          {
-            label: 'Delete Asset',
-            key: 'delete_asset',
-          },
-        ]}
-        data={this.state.displayAssetsList}
-      />
+      <div>
+        <Table
+          columns={[
+            {
+              label: 'Name',
+              key: 'display_name',
+            },
+            {
+              label: 'Type',
+              key: 'content_type',
+            },
+            {
+              label: 'Date Added',
+              key: 'date_added',
+            },
+            {
+              label: 'Delete Asset',
+              key: 'delete_asset',
+            },
+          ]}
+          data={this.state.displayAssetsList}
+        />
+        {this.renderModal()}
+      </div>
     );
   }
 }
 
 AssetsTable.propTypes = {
   assetsList: PropTypes.arrayOf(PropTypes.object).isRequired,
+  assetsParameters: PropTypes.objectOf(
```

Comment:
assetsParameters is necessary because it's needed by the deleteAsset action creator (it contains the courseID of the course in which the asset exists), which is dispatched from the deleteAsset function. I had to add it to the AssetTable to be able to dispatch the action creator from deleteAsset. Is there a better way? It seemed a little out of place here.

Reply:
If you're still wondering about this one, I can take a look at how this is set up more today, and see if I can remember if there's another way to handle it.